workers/task-driver/src/utils/order_states.rs

```diff
@@ -35,6 +35,9 @@ pub async fn record_order_fill(
     metadata.record_partial_fill(match_res.base_amount, price);
     if metadata.data.amount == metadata.total_filled() {
         metadata.state = OrderState::Filled;
+        // If the order was completely filled, remove it from the local orders set
+        // as it is no longer matchable
+        state.remove_local_order(&order_id).await?;
```

Comment:
this will only happen on the executor, perhaps we should make this part of a state application. Otherwise, it's not entirely important this happens, so maybe we just delete this one

Reply:
good point, i'll try to do this in the applicator